proposals/aggregate.json

```diff
@@ -0,0 +1,135 @@
+{
+    "id": "aggregate",
+    "summary": "Aggregation based on general intervals",
+    "description": "Computes a aggregation based on an array of intervals.\n\nThe computed values will be projected to the labels. If no labels are specified, the lower value of the interval will be used as label for the corresponding values. In case of a conflict (i.e. the user-specified values for the lower values of the intervals are not distinct), the user-defined labels must be specified in the parameter `labels` as otherwise a `DistinctDimensionLabelsRequired` exception would be thrown. The number of user-defined labels and the number of intervals need to be equal.",
+    "categories": [
+        "cubes",
+        "aggregate"
+    ],
+    "experimental": true,
+    "parameters": [
+        {
+            "name": "data",
+            "description": "A data cube.",
+            "schema": {
+                "type": "object",
+                "subtype": "datacube"
+            }
+        },
+        {
+            "name": "intervals",
+            "description": "Left-closed intervals, which are allowed to overlap. Each interval in the array has exactly two elements:\n\n1. The first element is the lower value of the interval. The specified value is **included** in the interval.\n2. The second element is the upper value of the temporal interval. The specified value is **excluded** from the interval.\n\nThe second element must always be greater than the first element. Otherwise, an `ExtentEmpty` exception is thrown.",
+            "schema": {
+                "type": "array",
+                "minItems": 1,
+                "items": {
+                    "type": "array",
+                    "uniqueItems": true,
+                    "minItems": 2,
+                    "maxItems": 2,
+                    "items": {
+                        "type": "number"
```

Comment:
So intervals are only based on numbers? That implies that this process only works along a spatial dimension in practice?
Other dimensions don't have numeric labels: band dimensions have strings and temporal dimensions have date/datetime (subtype of string)

Reply:
The order of string values are not really well-defined. As such, yes only numerical. Dimensions can have numerical values though, we have a couple of processes that just index the labels if no further information is given (e.g. apply_dimension).